File: src/Time.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author Fagundes
 */
public class Time {
    private String nome;
    private int pontos = 0;
    private int vitorias = 0;
    private int empates = 0;
    private int derrotas = 0;
    private int golsPro = 0;
    private int golsContra = 0;
    private int qtdJogos = 0;

    public Time(String nome) {
        this.nome = nome;
    }

    public void registrarJogo(int golsFeitos, int golsSofridos) {
        this.qtdJogos++;
        this.golsPro += golsFeitos;
        this.golsContra += golsSofridos;

        if (golsFeitos > golsSofridos) {
            this.vitorias++;
            this.pontos += 3;
        } else if (golsFeitos == golsSofridos) {
            this.empates++;
            this.pontos += 1;
        } else {
            this.derrotas++;
        }
    }

    public String getNome() { 
        return nome; 
    }
    public int getPontos() {
        return pontos;
    }
    public int getVitorias() {
        return vitorias; 
    }
    public int getEmpates() {
        return empates; 
    }
    public int getDerrotas() {
        return derrotas; 
    }
    public int getGolsPro() {
        return golsPro; 
    }
    public int getGolsContra() {
        return golsContra; 
    }
    public int getJogosDisputados() {
        return qtdJogos; 
    }
    public int getSaldoGols() {
        return this.golsPro - this.golsContra; 
    }
}

File: brasileirao_py/modelos/time.py
from .base import ModeloBase

class Time(ModeloBase):
    
    def __init__(self, nome):
        self.nome = nome
        self.pontos = 0
        self.vitorias = 0
        self.empates = 0
        self.derrotas = 0
        self.gols_marcados = 0
        self.gols_sofridos = 0
        self.partidas_jogadas = 0
        self.historico_pontos = [0]
    
    def registrar_partida(self, gols_feitos, gols_sofridos):
        self.partidas_jogadas += 1
        self.gols_marcados += gols_feitos
        self.gols_sofridos += gols_sofridos
        
        if gols_feitos > gols_sofridos:
            self.vitorias += 1
            self.pontos += 3
        elif gols_feitos == gols_sofridos:
            self.empates += 1
            self.pontos += 1
        else:
            self.derrotas += 1
            
        self.historico_pontos.append(self.pontos)
    
    @property
    def saldo_gols(self):
        return self.gols_marcados - self.gols_sofridos
    
    @property
    def aproveitamento(self):
        if self.partidas_jogadas == 0:
            return 0.0
        return (self.pontos / (self.partidas_jogadas * 3)) * 100
    
    def para_dict(self):
        return {
            'nome': self.nome,
            'pontos': self.pontos,
            'vitorias': self.vitorias,
            'empates': self.empates,
            'derrotas': self.derrotas,
            'gols_marcados': self.gols_marcados,
            'gols_sofridos': self.gols_sofridos,
            'partidas_jogadas': self.partidas_jogadas,
            'saldo_gols': self.saldo_gols,
            'aproveitamento': self.aproveitamento
        }